cmake/CMakeLists.txt

```diff
@@ -960,7 +960,11 @@ if (WIN32)
         # issued by thrust nonstandard extension used: nameless struct/union
         list(APPEND ORT_WARNING_FLAGS "/wd4201")
         # warning C4800: Implicit conversion from 'X' to bool. Possible information loss
-        list(APPEND ORT_WARNING_FLAGS "/w34800")
+        if (onnxruntime_USE_OPENVINO)
```

Comment:
where does this warning get emitted? should this have only been added for openvino targets?
i.e. follow this pattern as example:
https://github.com/microsoft/onnxruntime/blob/908765c860834091c36a5fe73d862e48097fb364/cmake/CMakeLists.txt#L1675
and
https://github.com/microsoft/onnxruntime/blob/908765c860834091c36a5fe73d862e48097fb364/cmake/onnxruntime_providers.cmake#L602

Reply:
@jywu-msft Hi George. I think, very recently this level 3 warning flag has been added by default /w34800 for WIN32 builds. so, we had to put a check here, to disable it for OpenVINO EP.

we did even try disabling this warning (/wd4800) at a later stage in the cmake and it wasn't effective and not working.

I did even try the way as commented above by you  where I set the warnings to be disabled in cmake and called it out in onnxruntime_providers.cmake for OpenVINO EP and it didn't work either.

Reply:
> 

ok. thanks for checking. i'll take a look at it later as well.